scripts/pre_commit_linter_test.py

```diff
@@ -0,0 +1,292 @@
+    def setUp(self):
+        super(JsAndTsLintTests, self).setUp()
+        self.other_lints_swap = self.swap(
+            pre_commit_linter.OtherLintChecksManager, 'perform_all_lint_checks',
+            mock_perform_all_lint_checks)
+        self.check_codeowner_swap = self.swap(
+            pre_commit_linter, 'check_codeowner_file',
+            mock_check_codeowner_file)
+
+    def test_valid_js_file(self):
+        with self.print_swap, self.other_lints_swap, self.check_codeowner_swap:
+            pre_commit_linter.main(args=['--path=%s' % VALID_JS_FILEPATH])
+        self.assertTrue(all_checks_passed(self.linter_stdout))
+        self.assertTrue(
+            appears_in_linter_stdout(
+                ['SUCCESS   1 JavaScript and Typescript files linted'],
+                self.linter_stdout)
+            )
+
+    def test_invalid_js_file(self):
+        with self.print_swap, self.sys_swap:
+            with self.other_lints_swap, self.check_codeowner_swap:
+                pre_commit_linter.main(args=['--path=%s' % INVALID_JS_FILEPATH])
+        self.assertFalse(all_checks_passed(self.linter_stdout))
+        self.assertTrue(appears_in_linter_stdout(
+            ['35:63  error  Closing curly brace should be on the same line '
+             'as opening curly brace or on the line after the previous block  '
+             'brace-style',
+             'FAILED    1 JavaScript and Typescript files'],
+            self.linter_stdout))
+
+    def test_valid_ts_file(self):
+        with self.print_swap, self.other_lints_swap, self.check_codeowner_swap:
+            pre_commit_linter.main(args=['--path=%s' % VALID_TS_FILEPATH])
+        self.assertTrue(all_checks_passed(self.linter_stdout))
+        self.assertTrue(
+            appears_in_linter_stdout(
+                ['SUCCESS   1 JavaScript and Typescript files linted'],
+                self.linter_stdout)
+            )
+
+    def test_invalid_ts_file(self):
+        with self.print_swap, self.sys_swap:
+            with self.other_lints_swap, self.check_codeowner_swap:
+                pre_commit_linter.main(args=['--path=%s' % INVALID_TS_FILEPATH])
+        self.assertFalse(all_checks_passed(self.linter_stdout))
+        self.assertTrue(appears_in_linter_stdout(
+            ['21:3  error  Extra space after key \'value\'  key-spacing',
+             'FAILED    1 JavaScript and Typescript files'],
+            self.linter_stdout))
+
+
+class GeneralLintTests(LintTests):
+    """"Tests all other general lint functions."""
+    def setUp(self):
+        super(GeneralLintTests, self).setUp()
+        self.other_lints_swap = self.swap(
+            pre_commit_linter.OtherLintChecksManager, 'perform_all_lint_checks',
+            mock_perform_all_lint_checks)
+        self.js_ts_lint_swap = self.swap(
+            pre_commit_linter.JsTsLintChecksManager, 'perform_all_lint_checks',
+            mock_perform_all_lint_checks)
+
+    # TODO(#7590): Add codeowner checks.
```

Comment:
Let's remove this section along with the section below since it's already in the issue tracker.

Reply:
@kevinlee12 Just to clarify, the entire GeneralLintTests class correct?